packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp

```diff
@@ -333,8 +334,8 @@ void DistributorActor::doPostsNbrAllToAllVImpl(const DistributorPlan &plan,
   auto& [importStarts, importLengths] = importSubViewLimits;
   auto& [exportStarts, exportLengths] = exportSubViewLimits;
 
-  for (size_t pp = 0; pp < plan.getNumSends(); ++pp) {
-    sdispls[plan.getProcsTo()[pp]] = exportStarts[pp];
+  for (size_t pp = 0; pp < numSends; ++pp) {
+    sdispls[pp] = exportStarts[pp];
```

Comment:
I think we can have a problem here where numSends > the size of the MPI communicator (@csiefer2, "slow-path", is this still a concern?)